File: demo2/contrat/src/main/java/com/tarmiz/contrat/model/Contrat.java
package com.tarmiz.contrat.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Entity
@Table(name = "contrats")
public class Contrat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String police;
    private LocalDate dateSignature;
    private LocalDate dateExpiration;
    private Long  assureId;

}

File: src/app/class/contrat.ts
export class Contrat {

    id!: number;
    police: string;
    dateSignature: Date;
    dateExpiration: Date;
    // AssureId: number;
    assureId: number;
  
    constructor() {
      this.police = '';
      this.dateSignature = new Date();
      this.dateExpiration = new Date();
      // this.AssureId = 0;
      this.assureId = 0;
    }
}
